include/seqan3/alphabet/composite/alphabet_variant.hpp

```diff
@@ -163,6 +163,25 @@ namespace seqan3
  *     these are constructible/assignable/equality-comparable with;
  *   * only convertible to its alternatives through the member function convert_to() (which can throw!)
  *
+ * ### The `char` representation of an alphabet_variant
+ *
+ * Part of the seqan3::Alphabet concept requires that the alphabet_variant provides a char representation in addition
+ * to the rank representation. For an object of seqan3::alphabet_variant, the `to_char()` member function will always
+ * return the same character as if invoked on the respective alternative.
+ * In contrast, the `assign_char()` member function might be ambiguous between the alternative alphabets in a variant.
+ *
+ * For example, assigning a '!' to seqan3::dna15 resolves to an object of rank 8 with char representation 'N' while
+ * assigning '!' to seqan3::gap always resolves to rank 0, the gap symbol itself ('-'_gap).
+ * We tackle this ambiguousness by **defaulting unknown characters to the representation of the first alternative**
+ * (e.g. `alphabet_variant<dna15, gap>{}.assign_char('!')` resolves to rank 8, representing `N`_dna15).
+ *
+ * On the other hand, two alternative alphabets might have the same char representation (e.g if
+ * you combine dna4 with dna5, 'A', 'C', 'G' and 'T' are ambiguous).
+ * We tackle this ambiguousness by **always choosing the first valid char representation** (e.g.
+ * `alphabet_variant<dna4, dna5>{}.assign_char('A')` resolves to rank 0, representing an `A`_dna4).
+ *
+ *
```

Comment:
To explicitly assign via the character representation of a specific alphabet, assign to that type first and then assign to the variant, e.g.
```cpp
alphabet_variant<dna4, dna5> var;
var.assign_char('A');             // will be in the "dna4-state"
var = 'A'_dna5;                   // will be in the "dna5-state"
```